File: src/sunstone/public/app/tabs/vms-tab/panels/template.js
/* -------------------------------------------------------------------------- */
/* Copyright 2002-2016, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */

define(function(require) {
  /*
    DEPENDENCIES
   */

  var Locale = require('utils/locale');
  var Humanize = require('utils/humanize');

  /*
    CONSTANTS
   */

  var TAB_ID = require('../tabId');
  var PANEL_ID = require('./template/panelId');
  var RESOURCE = "VM"
  var XML_ROOT = "VM"

  /*
    CONSTRUCTOR
   */

  function Panel(info) {
    this.title = Locale.tr("Template");
    this.icon = "fa-file-o";

    this.element = info[XML_ROOT];

    return this;
  };

  Panel.PANEL_ID = PANEL_ID;
  Panel.prototype.html = _html;
  Panel.prototype.setup = _setup;

  return Panel;

  /*
    FUNCTION DEFINITIONS
   */

  function _html() {
    return '<div class="row">'+
      '<div class="large-12 columns">'+
        '<table class="dataTable">'+
          '<thead>'+
            '<tr>'+
              '<th colspan="2">' + Locale.tr("Template") + '</th>'+
            '</tr>'+
          '</thead>'+
          '<tbody>'+
            Humanize.prettyPrintJSON(this.element.TEMPLATE) +
          '</tbody>'+
        '</table>'+
      '</div>'+
    '</div>'
  }

  function _setup(context) {
  }
});

File: src/oca/go/src/goca/schemas/datastore/keys/template.go
/* -------------------------------------------------------------------------- */
/* Copyright 2002-2024, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/*--------------------------------------------------------------------------- */

package keys

// Template is a type for datastore template keys.
type Template string

const (
	Name                   Template = "NAME"
	Type                   Template = "TYPE"
	DSMAD                  Template = "DS_MAD"
	TMMAD                  Template = "TM_MAD"
	RestrictedDirs         Template = "RESTRICTED_DIRS"
	SafeDirs               Template = "SAFE_DIRS"
	NoDecompress           Template = "NO_DECOMPRESS"
	LimitTransferBW        Template = "LIMIT_TRANSFER_BW"
	DatastoreCapacityCheck Template = "DATASTORE_CAPACITY_CHECK"
	LimitMB                Template = "LIMIT_MB"
	BridgeList             Template = "BRIDGE_LIST"
	StagingDir             Template = "STAGING_DIR"
	Driver                 Template = "DRIVER"
	CompatibleSysDs        Template = "COMPATIBLE_SYS_DS"
)
